sherpa/astro/ui/utils.py

```diff
@@ -4733,6 +4733,162 @@ def pack_table(self, id=None):
+            These correspond to the Elow and Ehigh columns (represented
+            by the ENERG_LO and ENERG_HI columns of the MATRIX block) of
+            the OGIP standard.
+        startchan : int, optional
+            The starting channel number: expected to be 0 or 1 but this is
+            not enforced.
+        e_min, e_max : None or array, optional
+            The E_MIN and E_MAX columns of the EBOUNDS block of the
+            RMF.
+        ethresh : number or None, optional
+            Passed through to the DataARF call. It controls whether
+            zero-energy bins are replaced.
+        fname : str, optional
+            The name of the image file generated from the standard RMF file
+            with the CIAO tool rmfimg
+        name : str
+            The name of the data set
+        Returns
+        -------
+        rmf : DataRMF instance
+        Notes
+        -----
+        I do not think I have the startchan=0 case correct (does the
+        f_chan array have to change?).
+        """
+
+        def create_delta_rmf(rmflo, rmfhi, startchan=1,
+                             e_min=None, e_max=None, ethresh=None,
+                             name='delta-rmf'):
+            assert rmflo.size == rmfhi.size
+            assert startchan >= 0
+
+            # Set up the delta-function response.
+            # TODO: should f_chan start at startchan?
+            #
+            nchans = rmflo.size
+            matrix = numpy.ones(nchans, dtype=numpy.float32)
+            dummy = numpy.ones(nchans, dtype=numpy.int16)
+            f_chan = numpy.arange(1, nchans + 1, dtype=numpy.int16)
+
+            return sherpa.astro.data.DataRMF(name, detchans=nchans,
+                                             energ_lo=rmflo, energ_hi=rmfhi,
+                                             n_grp=dummy, n_chan=dummy,
+                                             f_chan=f_chan, matrix=matrix,
+                                             offset=startchan,
+                                             e_min=e_min, e_max=e_max,
+                                             ethresh=ethresh)
+
+        def create_non_delta_rmf(rmflo, rmfhi, fname, startchan=1,
+                                 e_min=None, e_max=None, ethresh=None,
+                                 name='delta-rmf'):
+            assert rmflo.size == rmfhi.size
+            assert startchan >= 0
+            assert os.path.isfile(fname)
+
+
+            # Set up the delta-function response.
+            # TODO: should f_chan start at startchan?
+            #
+            nchans = rmflo.size
+            def calc_grp_chan_matrix(fname):
+                try:
+                    data, filename = \
+                        sherpa.astro.io.backend.get_image_data(fname)
+                    matrix = data['y']
+                    n_grp = []
+                    n_chan = []
+                    f_chan = []
+                    for row in (matrix > 0):
```

Comment:
This condition does not seem to make sense, but I might be wrong. However, since there are no automated tests, I can't run it in a debugger.